Separate code exercise descriptions (#74)
sourcegraph/careers: job-descriptions/code-exercise-go.md

# Go code exercise

We designed this exercise to measure your understanding of HTTP requests, concurrency, and error handling.

> - You will choose a two hour timeframe to independently work on the exercise. You can choose any day and time; we just schedule an email.

Author: Despite trying to make it painfully clear that candidates can literally choose any time they want, they still routinely send back multiple times and ask which works best for me. Going to update the language to the following:

```suggestion
- You will choose a two hour timeframe to independently work on the exercise. You can choose any day and time that works for you; there are no scheduling constraints.
```
